### PR TITLE
Block Library: Remove `post-` prefix from Comment Template inner blocks

### PR DESCRIPTION
**Pull request by gziolo**
## Description

The idea of this refactor is to use consistent code names for all inner blocks used with the Comment Template block that has the internal name `core/comment-template`. We also have the Comments Query Loop block with the internal name `core/comments-query-loop`. The next step after we landed in https://github.com/WordPress/gutenberg/pull/36032 changes in the UI for all comments related block to stop using the Post keyword is to propagate the same change to folder names, file names, and internal block names. The changes proposed for block names and their matching folder names:

- `core/post-comment-author` -> `core/comment-author-name`
- `core/post-comment-author-avatar` -> `core/comment-author-avatar`
- `core/post-comment-content` -> `core/comment-content`
- `core/post-comment-date` -> `core/comment-date`
- `core/post-comment-edit` -> `core/comment-edit-link`
- `core/post-comment-reply-link` -> `core/comment-reply-link` 

## Review notes

It looks like Git wrongly mapped some file renames so the visual diff sometimes shows two different files in the comparison view 🙃 

Git wasn't also able to match some renamed test block fixture files ...

## How has this been tested?

All unit tests pass locally when running `npm run test-unit` - it covers changes to test block fixtures.
All CI checks pass.

I confirmed in the site editor that the deprecated Post Comment block and Comments Query Loop block continue to work as before with all the renamed blocks.

### Unreleased blocks get invalid

It's expected behavior. The blocks that existed for a few past releases get mapped from old to new names in the block parser.

Screen Shot 2021-11-03 at 11 17 38

### Deprecated Post Comment block

Screen Shot 2021-11-03 at 11 16 44

### Comment Query Loop block

Screen Shot 2021-11-03 at 11 18 47
Screen Shot 2021-11-03 at 11 19 18

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/icons/src/index.js [hunk 1]
```diff
@@ -49,8 +49,10 @@ export { default as color } from './library/color';
 export { default as column } from './library/column';
 export { default as columns } from './library/columns';
 export { default as comment } from './library/comment';
-export { default as commentAuthor } from './library/comment-author';
-export { default as postCommentContent } from './library/post-comment-content';
```
Comment (gziolo) [inline packages/icons/src/index.js:53]: `commentAuthor` and `postCommentContent` are already published to npm:
https://unpkg.com/@wordpress/icons@6.0.1/build/index.js

What's interesting is they don't exist in https://unpkg.com/@wordpress/icons@6.0.0/build/index.js so they were added as bug fixes.

We will have to call it a breaking change in the CHANGELOG of the package.

Comment (gziolo) [reply packages/icons/src/index.js:53]: I changed my mind. Given that those new icons were added specifically for core blocks I doubt they are used anywhere else. They weren't included in the CHANGELOG before or exposed in the documentation. Let's list new icons in the CHANGELOG as new features as proposed in https://github.com/WordPress/gutenberg/pull/36171/commits/de40719e7e22810143749f0aa4cb19dd0d745595.

File: packages/block-library/src/comment-author-avatar/index.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /**
  * WordPress dependencies
  */
-import { postCommentReplyLink as icon } from '@wordpress/icons';
+import { commentAuthorAvatar as icon } from '@wordpress/icons';
```
Comment (cbravobernal) [inline packages/block-library/src/comment-author-avatar/index.js:4]: Is this a git error, right?

Comment (gziolo) [reply packages/block-library/src/comment-author-avatar/index.js:4]: Yes, I left a note in the description:

> It looks like Git wrongly mapped some file renames so the visual diff sometimes shows two different files in the comparison view 🙃
>
> Git wasn't also able to match some renamed test block fixture files ...

File: packages/block-library/src/comment-template/edit.js [hunk 3]
```diff
@@ -15,9 +15,10 @@ import { Spinner } from '@wordpress/components';
 import { store as coreStore } from '@wordpress/core-data';
 
 const TEMPLATE = [
-	[ 'core/post-comment-author' ],
-	[ 'core/post-comment-date' ],
-	[ 'core/post-comment-content' ],
+	[ 'core/comment-author-name' ],
```
Comment (cbravobernal) [inline packages/block-library/src/comment-template/edit.js:19]: should we add here `core/post-comment-author-avatar`? Or is yet too soon to do it.

File: packages/block-library/src/post-comment/edit.js [hunk 4]
```diff
@@ -8,15 +8,18 @@ import { blockDefault } from '@wordpress/icons';
...
+	'core/comment-content',
+	'core/comment-date',
+	'core/comment-edit-link',
+	'core/comment-reply-link',
 ];
 const TEMPLATE = [
-	[ 'core/post-comment-content' ],
-	[ 'core/post-comment-author' ],
-	[ 'core/post-comment-reply-link' ],
+	[ 'core/comment-author-name' ],
```
Comment (cbravobernal) [inline packages/block-library/src/post-comment/edit.js:20]: should we add here core/post-comment-author-avatar? Or is yet too soon to do it.

**Review by cbravobernal — APPROVED** [review]
Looks good to me. Nice job!

File: packages/block-library/src/comment-template/edit.js (hunk 3 above)
Comment (gziolo) [reply packages/block-library/src/comment-template/edit.js:19]: We need some initial design. I can include all available blocks for now 😃

Comment (gziolo) [reply packages/block-library/src/comment-template/edit.js:19]: Done in 2aaefc6da434ef7642e0bc4557a818a75b7cc3ad.

File: packages/block-library/src/post-comment/edit.js (hunk 4 above)
Comment (gziolo) [reply packages/block-library/src/post-comment/edit.js:20]: Done in 2aaefc6da434ef7642e0bc4557a818a75b7cc3ad.

**Pull request merged**